third-party/realdds/include/realdds/dds-stream-profile.h

```diff
@@ -87,6 +94,10 @@ class dds_stream_profile
     }
 
 public:
+    std::shared_ptr< dds_stream_base > stream() const { return _stream.lock(); }
+    // This is for initialization and is called from dds_stream_base only!
+    void init_stream( std::weak_ptr< dds_stream_base > const & stream );

```

Comment:
Can we enforce calling from `dds_stream_base` only? Maybe making it a friend and the function protected

Reply:
I thought of that and decided against it.
If you pass a profile that was already a part of another stream, it won't let you. You can really only call this function once, and adding it to a stream automatically calls it.

Reply:
It is good that dds_stream_base will call this automatically but can we protect agains calls from other classes?

Reply:
We do -- a profile cannot be added to a stream if it already has a stream